make/modules/java.desktop/lib/AwtLibraries.gmk

```diff
@@ -172,10 +172,16 @@ ifeq ($(call isTargetOs, windows macosx), false)
   # static libraries cause linking errors due to duplicate symbols.
   LIBAWT_HEADLESS_STATIC_EXCLUDE_OBJS := systemScale.o
 
+  ifneq ($(ENABLE_HEADLESS_ONLY), true)
+    # We cannot link with both awt_headless and awt_xawt at the same time
```

Comment:
Just a note on that. It's doable to link with both awt_headless and awt_xawt with some work. I did some quick experiments on that during the initial investigation for hermetic/static Java.

Reply:
That would require quite some work then..! The two libraries are meant as exclusive complements to each other -- they both implement the same "entry points", but in different ways -- one with X11 support, and one without. For other reasons (outside of static launcher reasons) I'd like to see some refactoring in how this is implemented, but that is completely outside this discussion.

For the static launcher scenario, I can't even see the point of trying to include both? What would you accomplish by that? 

The entire point of having two libraries is that you want to be able to have full workstation capabilities, but then be dependent on the X11 libraries, or have limited capabilities, but skip the X11 dependency.

Reply:
My initial understanding was that the libawt_headless was mostly as subset of libawt_xawt, which made it possible to statically link both the headless and headful natives. Completely agree that it's outside of the current scope.